### DISCUSSION TITLE
quick, simple type conversion question

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#12816] [Using MicroPython] [answered]
I can convert an integer (eg 0x32) to a string ('0x32') easily
```
>>> i=0x32; s=str(i); print(repr(i), type(i), repr(s), type(s))
50 <class 'int'> '50' <class 'str'>
```
but I can't think of a way to do the reverse, ie change '0x32' into 0x32. Curious if someone knows how?

**Comment by andrewleech** [comment, marked as answer]
There's a few ways but these are the first ways I thought of to convert both directions
```
>>> i = 0x32
>>> i
50
>>> hex(i)
'0x32'
>>> int('0x32', 0)
50
```

**Reply by jimmo** [reply]
Yep this is the best way. The `0` is a bit magical here, it means "detect the base". You can also explicitly do `int('0x32', 16)` which will fail if it doesn't start with "0x".

**Reply by andrewleech** [reply]
Or if you want to convert decimal instead of hex:
```
>>> int('50', 0)
50
>>> int('50', 10)
50
```

**Comment by steveh127** [comment]
Or simply int('0x32') no need for the  0 or 16. The conversion is simple in either direction.

**Comment by kjm1102** [comment]
When  I convert integer to string upython throws in a hex-to-decimal conversion for both types
```
>>> i=0x32; s=str(i); print(repr(i), type(i), repr(s), type(s))
50 <class 'int'> '50' <class 'str'>
```
but when I convert string to integer
```
>>> s='0x32'; i=int(s); print(repr(s), type(s), repr(i), type(i))
'0x32' <class 'str'> 50 <class 'int'>
```
I only get the hex-to-decimal conversion for one of the types. Does anyone know why that is?

**Reply by andrewleech** [reply]
An integer number doesn't have a hex or decimal conversion, a number is just a number. It's only the _display_ of the number that can be in hex or decimal form, the number itself is always the same.

If you want to display an integer in hex (string) form, you can use the `hex(i)` function I used in my first example.

**Reply by robert-hh** [reply]
Your code asks python to print a string, and that is done. By chance, this string represents a hex number, but it could as well be any text.
